packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts

```diff
@@ -165,6 +178,60 @@ export class MongoDBDataProvider<Type = any, GraphbackContext = any> implements
     return document;
   }
 
+  protected getIndexFields(baseType: GraphQLObjectType): IndexSpecification[] {
+    const res: IndexSpecification[] = [];
+    const fields = baseType.getFields();
+    Object.keys(fields).forEach((k: string) => {
+      const field = fields[k];
+
+      // Add custom Index if found e.g. @db(index: 1)
+      const customIndex = this.getCustomIndex(field);
+      if (customIndex !== undefined) {
+        res.push(customIndex);
+
+        // To prevent adding mutliple index on same field
+        return;
+      }
+      
+      // Add Index on relation fields
+      const relationIndex = this.getRelationIndex(field);
+      if (relationIndex !== undefined) {
+        res.push(relationIndex);
+      }
+
+    })
+    
+    return res;
+  }
+
+  private getCustomIndex(field: GraphQLField<any, any>): IndexSpecification {
+    const otherAnnotations = parseMetadata('db', field.description) as any;
```

Comment:
Let's not use the `db` namespace, see https://github.com/aerogear/graphback/issues/1436#issuecomment-644045719